rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -463,13 +463,25 @@ void SqliteStorage::prepare_for_reading()
 
 void SqliteStorage::fill_topics_and_types()
 {
-  auto statement = database_->prepare_statement(
-    "SELECT name, type, serialization_format FROM topics ORDER BY id;");
-  auto query_results = statement->execute_query<std::string, std::string, std::string>();
+  if (database_->field_exists("topics", "offered_qos_profiles")) {
+    auto statement = database_->prepare_statement(
+      "SELECT name, type, serialization_format, offered_qos_profiles FROM topics ORDER BY id;");
+    auto query_results =
+      statement->execute_query<std::string, std::string, std::string, std::string>();
 
-  for (auto result : query_results) {
-    all_topics_and_types_.push_back(
-      {std::get<0>(result), std::get<1>(result), std::get<2>(result), ""});
+    for (auto result : query_results) {
+      all_topics_and_types_.push_back(
+        {std::get<0>(result), std::get<1>(result), std::get<2>(result), std::get<3>(result)});
+    }
+  } else {
+    auto statement = database_->prepare_statement(
+      "SELECT name, type, serialization_format FROM topics ORDER BY id;");
+    auto query_results = statement->execute_query<std::string, std::string, std::string>();
+
+    for (auto result : query_results) {
+      all_topics_and_types_.push_back(
+        {std::get<0>(result), std::get<1>(result), std::get<2>(result), ""});
+    }
```

Comment:
Note for reviewers. This if-else statement was missing on humble because we didn't backported https://github.com/ros2/rosbag2/pull/1722. However, it is required to correctly fill topics and types map with `offered_qos_profiles`.